receiver/splunkhecreceiver/receiver.go

```diff
@@ -295,18 +389,45 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, err)
 		return
 	}
+
 	consumerErr := r.logsConsumer.ConsumeLogs(ctx, ld)
 
 	_ = bodyReader.Close()
 
 	if consumerErr != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, slLen, consumerErr)
 	} else {
-		r.writeSuccessResponse(ctx, resp, ld.LogRecordCount())
+		if len(partitionID) > 0 && r.ackExt != nil {
+			r.writeSuccessResponseWithAck(ctx, resp, ld.LogRecordCount(), partitionID)
+		} else {
+			r.writeSuccessResponseWithoutAck(ctx, resp, ld.LogRecordCount())
+		}
 		r.obsrecv.EndLogsOp(ctx, metadata.Type.String(), slLen, nil)
 	}
 }
 
+func (r *splunkReceiver) extractChannelHeader(req *http.Request) (string, bool) {
+	headers, ok := req.Header[splunk.HTTPSplunkChannelHeader]
+	if ok {
+		return headers[0], true
+	}
+	return "", false
+}
+
+func (r *splunkReceiver) validateChannelHeader(partitionID string) error {
```

Comment:
What's the difference between a channel header and partition ID? If they're the same thing, why do we use two names?

Reply:
the partition is called a channel for hec only. However the ack extension might be used by other receiver as well so partition is the general name. I can make this consistent and just call it channel.

Reply:
That makes sense, thanks for clarifying. It should be fine to leave the naming as-is.